api_coverage.md

```diff
@@ -0,0 +1,108 @@
+| psa_raw_key_agreement            | :x:                                                                                                                    |
+
+## API support per-provider
+
+This section gives information on the operations and attributes currently supported by each provider. Only support for the operations marked as covered in the table(s) above is considered. Any option marked as supported is necessarily supported for all operations on which it is usable.
+
+### Operation support
+
+| Operation         | Mbed provider      | PKCS 11 provider   | TPM 2.0 provider   |
+| :---------------- | :----------------: | :----------------: | :----------------: |
+| Import key        | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| Generate key      | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| Export public key | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| Destroy key       | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| Sign hash         | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| Verify hash       | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+
+### Algorithm support
+
+#### Hash algorithms
+
+| Algorithm   | Mbed provider      | PKCS 11 provider   | TPM 2.0 provider   |
+| :---------- | :----------------: | :----------------: | :----------------: |
+| MD2         | :x:                | :x:                | :x:                |
+| MD4         | :x:                | :x:                | :x:                |
+| MD5         | :x:                | :x:                | :x:                |
+| RIPEMD160   | :x:                | :x:                | :x:                |
+| SHA-1       | :x:                | :x:                | :x:                |
+| SHA-224     | :x:                | :x:                | :x:                |
+| SHA-256     | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| SHA-384     | :x:                | :x:                | :x:                |
+| SHA-512     | :x:                | :x:                | :x:                |
+| SHA-512-224 | :x:                | :x:                | :x:                |
+| SHA-512-256 | :x:                | :x:                | :x:                |
+| SHA3-224    | :x:                | :x:                | :x:                |
+| SHA3-256    | :x:                | :x:                | :x:                |
+| SHA3-384    | :x:                | :x:                | :x:                |
+| SHA3-512    | :x:                | :x:                | :x:                |
+| Any hash    | :x:                | :x:                | :x:                |
+
+#### Asymmetric signing algorithms
+
+| Algorithm              | Mbed provider      | PKCS 11 provider   | TPM 2.0 provider   |
+| :--------------------- | :----------------: | :----------------: | :----------------: |
+| RSA PKCS 1v5 with hash | :white_check_mark: | :white_check_mark: | :white_check_mark: |
```

Comment:
Are we not only supporting the `raw` version?

Reply:
Don't think so, we check if the specified algorithm is RSA PKCS 1.5 with SHA256 in the providers. Also, the Raw one expects (I think?), that the client sends the ASN.1 header already attached